wweval/R/ms_fig4.R

```diff
@@ -604,46 +791,345 @@ make_fig4_avg_crps_over_time <- function(scores,
+#' containing the outputs of `scoringutils::score()` on samples plus metadata
+#' transformed into a tibble.
+#' @return a ggplot object that is a vertical facet of violin plots colored
+#' by model type and broken down my horizon
+#' @export
+make_fig4_rel_wis_over_time <- function(wis_scores) {
+  scores_overall <- wis_scores |>
+    dplyr::mutate(
+      horizon = "overall"
+    )
+
+  relative_wis <- scores_overall |>
+    data.table::as.data.table() |>
+    scoringutils::summarise_scores(by = c(
+      "forecast_date",
+      "location",
+      "model",
+      "horizon"
+    )) |>
+    tidyr::pivot_wider(
+      names_from = model,
+      values_from = interval_score,
+      id_cols = c("horizon", "forecast_date", "location")
+    ) |>
+    dplyr::mutate(
+      rel_wis = ww / hosp
+    ) |>
+    dplyr::filter(!is.na(rel_wis)) |>
+    order_locations(score_name = "rel_wis")
+
+
+  colors <- plot_components()
+  date_lims <- c(range(wis_scores$forecast_date))
+
+  p <- ggplot(relative_wis) +
+    tidybayes::stat_dotsinterval(
+      aes(
+        x = as.factor(forecast_date), y = rel_wis,
+        fill = horizon
+      ),
+      point_interval = "mean_qi",
+      alpha = 0.5,
+      position = position_dodge(width = 0.75),
+      show.legend = FALSE
+    ) +
+    geom_hline(aes(yintercept = 1), linetype = "dashed") +
+    xlab("") +
+    ylab("Relative WIS") +
+    scale_y_continuous(trans = "log10") +
+    coord_cartesian(ylim = c(0.5, 2)) +
+    get_plot_theme(
+      x_axis_dates = TRUE,
+      y_axis_title_size = 8
+    ) +
+    scale_fill_manual(values = colors$horizon_colors) +
+    scale_color_manual(values = colors$horizon_colors)
+
+  return(p)
+}
+
+#' Make figure that stratifies scores by location across forecast dates
+#'
+#' @param wis_scores A tibble of scores by location, forecast date, date and model,
+#' containing the outputs of `scoringutils::score()` on samples plus metadata
+#' transformed into a tibble.
+#'
+#' @return A ggplot object containing plots of the distribution of relative
+#' WIS scores by location, across forecast dates, colored by location
+#' @export
+make_fig4_rel_wis_by_location <- function(wis_scores) {
+  scores_overall <- wis_scores |>
+    dplyr::mutate(
+      horizon = "overall"
+    )
+
+  relative_wis <- scores_overall |>
+    data.table::as.data.table() |>
+    scoringutils::summarise_scores(by = c(
+      "forecast_date",
+      "location",
+      "model",
+      "horizon"
+    )) |>
+    tidyr::pivot_wider(
+      names_from = model,
+      values_from = interval_score,
+      id_cols = c("horizon", "forecast_date", "location")
+    ) |>
+    dplyr::mutate(
+      rel_wis = ww / hosp
+    ) |>
+    dplyr::filter(!is.na(rel_wis)) |>
+    order_locations(score_name = "rel_wis")
+
+  colors <- plot_components()
+
+  p <- ggplot(relative_wis) +
+    tidybayes::stat_dotsinterval(
+      aes(
+        x = location, y = rel_wis,
+        fill = horizon
+      ),
+      point_interval = "mean_qi",
```

Comment:
```suggestion
      point_interval = "median_qi",
```

Reply:
Hmm I think I disagree -- we decide to show the geometric mean on the plot (which is log transformed mean) not the median to maintain properness